util/runner.zig

```diff
@@ -101,7 +101,7 @@ pub fn next(self: *Runner, reading: Reading) Error!?Step {
                 var N: usize = @intCast((st.iteration_loops * st.time_budget_ns) / st.elapsed_ns);
                 // check that N doesn't go out of bounds
                 if (N == 0) N = 1;
-                if (N > DEFAULT_MAX_N_ITER) N = DEFAULT_MAX_N_ITER;
+                if (N > st.max_iterations) N = st.max_iterations;
```

Comment:
I think it should be `DEFAULT_MAX_N_ITER` *or* `st.max_iteration`, whatever is smaller, no?